libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -204,19 +204,39 @@ export class GRPCService implements Service {
   }
 
   private objectParser = (struct: Struct) => {
-    if (struct !== undefined) {
+    if (struct) {
       return Struct.toJson(struct);
-    } else {
-      throw new ParseError('Object value undefined or missing.');
     }
+    return {}
+  };
+
+  private booleanParser = (value: boolean) => {
+    if (value) {
+      return value;
+    }
+    return false
+  };
+
+  private stringParser = (value: string) => {
+    if (value) {
+      return value;
+    }
+    return ''
+  };
+
+  private numberParser = (value: number) => {
+    if (value) {
+      return value;
+    }
+    return 0
   };
 
   private async resolve<T extends FlagValue, Rq extends AnyRequest, Rs extends AnyResponse>(
     resolver: (request: AnyRequest) => UnaryCall<Rq, Rs>,
     flagKey: string,
     context: EvaluationContext,
     logger: Logger,
-    parser?: (struct: Struct) => PbJsonValue
+    parser?: (value: any) => any
```

Comment:
Can we restrict these types a bit more?

It can be a union of types: `Struct | number | string...`. Generally, we shouldn't see explicit `any`s (`unknown` is safer if you really don't know the type)

Reply:
it was quite problematic here, but will give it another try. If it won't go, I will use `undefined`

Reply:
I am trying desperately but cannot manage to make it work without the use of `any`. The problem is the conversion of the object resolution, which in this case can be anything. Would you maybe have time to sync regarding this problem? I do not have that deep knowledge regarding JS/Typescript, so maybe I am missing some good manners how to resolve this here...

Reply:
The problem here is that `Struct` and `FlagValue` are not compatible types.

Reply:
we were working on this together with @thisthat but didn't manage to find a solution. For now we prefer to stay with the current solution